### DISCUSSION TITLE
exchange OAuth 1.0/ OAuth2.0

### DISCUSSION THREAD

**Discussion by hassan-abrar-south** [matin/garth#36] [General]
Hey,
i have a question if you could explain me thanks.
i have checked your code and you have consumer key and secret and you have stored it in s3.
but how did you give excess to your garmin to execute using oAuth 2.0. because when i give my keys they just fails with forbidden error.

tell me how to give permission thanks.
looking forward to the answer.

**Comment by matin** [comment]
The keys stored in S3 aren't mine. They're the keys used by the Connect Android app.

**Reply by zimengzhou1** [reply]
Just a small follow up - how did you get the keys from the Connect android app? Thanks
